### PR TITLE
fix bug in fastchess output when report_penta=false

### PR DESCRIPTION
**Pull request by gahtan-syarif**
output:
Screenshot 2024-04-13 162652
This is caused by fastchess output still displaying penta stats when penta option is set to false. the fix for this is to display trinomial stats instead when report penta is false. ive added the report_penta conditional in output_fastchess.cpp but idk which header i should include so i can use the boolean, can you help?

Edit: found it

### CODE REVIEW

**Comment by Disservin** [comment]
```diff
diff --git a/src/matchmaking/output/output_factory.hpp b/src/matchmaking/output/output_factory.hpp
index 220b5ab..055630a 100644
--- a/src/matchmaking/output/output_factory.hpp
+++ b/src/matchmaking/output/output_factory.hpp
@@ -2,19 +2,20 @@

 #include <matchmaking/output/output_cutechess.hpp>
 #include <matchmaking/output/output_fastchess.hpp>
+#include <types/tournament_options.hpp>

 namespace fast_chess {

 class OutputFactory {
    public:
-    [[nodiscard]] static std::unique_ptr<IOutput> create(OutputType type) {
-        switch (type) {
+    [[nodiscard]] static std::unique_ptr<IOutput> create(const options::Tournament& config) {
+        switch (config.output) {
             case OutputType::FASTCHESS:
-                return std::make_unique<Fastchess>();
+                return std::make_unique<Fastchess>(config.report_penta);
             case OutputType::CUTECHESS:
                 return std::make_unique<Cutechess>();
             default:
-                return std::make_unique<Fastchess>();
+                return std::make_unique<Fastchess>(config.report_penta);
         }
     }

diff --git a/src/matchmaking/output/output_fastchess.hpp b/src/matchmaking/output/output_fastchess.hpp
index da0171c..0296375 100644
--- a/src/matchmaking/output/output_fastchess.hpp
+++ b/src/matchmaking/output/output_fastchess.hpp
@@ -8,6 +8,8 @@ namespace fast_chess {

 class Fastchess : public IOutput {
    public:
+    Fastchess(bool report_penta = true) : report_penta_(report_penta) {}
+
     void printInterval(const SPRT& sprt, const Stats& stats, const std::string& first,
                        const std::string& second, int current_game_count) override {
         std::cout << "--------------------------------------------------\n";
@@ -128,6 +130,9 @@ class Fastchess : public IOutput {
     }

     void endTournament() override { std::cout << "Tournament finished" << std::endl; }
+
+   private:
+    bool report_penta_;
 };

 }  // namespace fast_chess
diff --git a/src/matchmaking/tournament/base/tournament.cpp b/src/matchmaking/tournament/base/tournament.cpp
index 66399af..de6b9df 100644
--- a/src/matchmaking/tournament/base/tournament.cpp
+++ b/src/matchmaking/tournament/base/tournament.cpp
@@ -20,7 +20,7 @@ BaseTournament::BaseTournament(const options::Tournament &config,
                                const std::vector<EngineConfiguration> &engine_configs) {
     tournament_options_ = config;
     engine_configs_     = engine_configs;
-    output_             = OutputFactory::create(config.output);
+    output_             = OutputFactory::create(config);
     book_               = OpeningBook(config.opening);
     cores_              = std::make_unique<affinity::AffinityManager>(config.affinity,
                                                          getMaxAffinity(engine_configs));
```

i'd do it like this

**Comment by Disservin** [comment]
and then in fastchess you can check `report_penta_`

**Comment by gahtan-syarif** [comment]
hmm i got it to compile but it doesnt seem to fix the problem
Screenshot 2024-04-13 203900
ill try your approach

File: src/matchmaking/output/output_fastchess.hpp [hunk 1]
```diff
@@ -19,73 +20,127 @@ class Fastchess : public IOutput {
...
-           << Elo::getLos(stats.penta_WW, stats.penta_WD, stats.penta_WL, stats.penta_DD,
-                          stats.penta_LD, stats.penta_LL)  //
-           << ", "                                         //
-           << "PairDrawRatio: "                            //
-           << Elo::getDrawRatio(stats.penta_WW, stats.penta_WD, stats.penta_WL, stats.penta_DD,
-                                stats.penta_LD, stats.penta_LL)  //
-           << "\n";
-
-        std::cout << ss.str() << std::flush;
+        options::Tournament tournament;
```
Comment (Disservin) [inline src/matchmaking/output/output_fastchess.hpp:23]: well ofc it doesnt work :D you just create a new tournament config here which has the default values. @gahtan-syarif

Comment (gahtan-syarif) [reply src/matchmaking/output/output_fastchess.hpp:23]: ahh i see, so it doesnt access the updated values

**Pull request closed without merging**